base/linalg/generic.jl

```diff
@@ -103,21 +103,33 @@ function generic_vecnorm1(x)
     return convert(T, sum)
 end
 
+function generic_vecnorm2(n, scale, ssq, x)
+    @inbounds for i = 1:n
+        if x[i] != 0
+            absxi = norm(x[i])
+            if scale < absxi
+                ssq = 1 + ssq*(scale/absxi)^2
+                scale = absxi
+            else
+                ssq = ssq + (absxi/scale)^2
+            end
+        end
+    end
+    return scale * sqrt(ssq)
+end
+function generic_vecnorm2{T<:Number}(x::AbstractVector{T})
+    scale = zero(norm(zero(T)))
+    ssq = one(sqrt((abs2(norm(zero(T))) + abs2(norm(zero(T))))/2))
```

Comment:
Is all this really needed just to set `ssq = 0` of the appropriate type?   Unless the compiler is clever enough to discard all of the extraneous calculations?